src/core/public/chrome/ui/header/collapsible_nav.tsx

```diff
@@ -119,15 +148,32 @@ export function CollapsibleNav({
   ...observables
 }: Props) {
   const navLinks = useObservable(observables.navLinks$, []).filter((link) => !link.hidden);
+  let customNavLink = useObservable(observables.customNavLink$, undefined);
+  if (customNavLink) {
+    customNavLink = { ...customNavLink, externalLink: true };
+  }
   const recentlyAccessed = useObservable(observables.recentlyAccessed$, []);
-  const customNavLink = useObservable(observables.customNavLink$, undefined);
+  const allNavLinks: CollapsibleNavLink[] = [...navLinks];
+  if (recentlyAccessed.length) {
+    allNavLinks.push(
+      ...recentlyAccessed.map((link) => createRecentChromeNavLink(link, navLinks, basePath))
+    );
+  } else {
+    allNavLinks.push(emptyRecentlyVisited);
```

Comment:
Instead of pushing empty recent visited link, perhaps we can just leave it as it is, so recent visited is not displayed if there was no recent visited links.

Reply:
Do we need to keep this behavior consistent with our current UI: